### PR TITLE
Contributing guidance

### LINKED ISSUE #30: Define contributing workflow
It would be useful to discuss best practices and collaborative workflow, the describe the final decision in `contributing.md` for future collaborators to use when getting started.

### PR DESCRIPTION
**Pull request by gadgil48**
Closes #30 . Feel free to suggest any improvements.

### CODE REVIEW

**Comment by cmacmackin** [comment]
This looks great Sanket. Just noticed one very small typo which I'll leave a note on.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
...
+3. Develop and implement the feature on this new branch and check that it works. Write the appropriate tests for the new feature and check that the tests (both new and old) pass.
+4. Pull from remote `master` to your local `master` branch and merge any updates into your local feature branch before pushing your feature branch to remote. This ensures that any merge conflicts can be handled before the pull request is created. (Remember to push the feature branch regularly to the remote repo to act as a backup and prevent a loss of code locally)
+5. Once the issue has been addressed and the remote version of the feature branch is up-to-date with the local version, create a pull request. Mention the issue number (**with** `#`) in the pull request description to link the pull request to the issue so that the issue can be closed when the pull request is merged.
+6. The options on the pull request will need to be decided on a per-case basis in terms of whether to delete the source branch and/or squash the commits upon merging.
+7. Assign a reviewer(someone other than yourself) to review and merge the pull request. Note that merging will be disabled by default if any tests fail.
```
Comment (cmacmackin) [inline CONTRIBUTING.md:8]: Missing space before the open parenthesis.

**Review by cmacmackin — APPROVED** [review]

**Review by elitherl — APPROVED** [review]
Looks good, the only issue I could see has already been pointed out

**Review by marcosertoli — APPROVED** [review]
Looks good to me Sanket.

**Pull request merged**